src/parse_type.rs

```diff
@@ -20,6 +20,31 @@ pub(crate) fn consume_declaration_name(tokens: &mut TokenIter) -> Ident {
     parse_any_ident(tokens, "declaration")
 }
 
+pub(crate) fn consume_bound(
+    tokens: &mut TokenIter,
+    mut end_predicate: impl FnMut(&TokenTree) -> bool,
+) -> Option<GenericBound> {
+    match tokens.peek().unwrap() {
+        TokenTree::Punct(punct) if punct.as_char() == ':' => {
+            let colon = punct.clone();
+            // consume ':'
+            tokens.next();
+
+            let bound_tokens = consume_stuff_until(tokens, end_predicate, false);
+
+            Some(GenericBound {
+                tk_colon: colon,
+                tokens: bound_tokens,
+            })
+        }
+        TokenTree::Punct(punct) if punct.as_char() == '>' => None,
```

Comment:
I'm a bit iffy on that line. It could probably be moved to `consume_generic_params` ?

Reply:
True, fixed.